frontends/mit-open/src/libs/mui.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react"
+import { createTheme } from "@mui/material/styles"
+import { LinkProps } from "@mui/material/Link"
+import {
+  Link as RouterLink,
+  LinkProps as RouterLinkProps,
+} from "react-router-dom"
+
+/**
+ * An adapter for MUI's link component to work with react-router.
+ *
+ * See https://mui.com/material-ui/guides/routing/ for more.
+ */
```

Comment:
This is the new part of this file, which was converted from ts to tsx for the react component.

Before, a `<Button href="internal site link" />` would do a full site reload. This makes the MUI link aware of react router, so now it routes like a SPA.